pkgs/development/compilers/c3c/default.nix

```diff
@@ -0,0 +1,32 @@
+{ llvmPackages
+, lib
+, fetchFromGitHub
+, cmake
+}:
+
+llvmPackages.stdenv.mkDerivation rec {
+  pname = "c3c";
+  version = "unstable-2021-07-26";
+
+  src = fetchFromGitHub {
+    owner = "c3lang";
+    repo = pname;
+    rev = "9f42ddb68d4b687c2afd8543fa4df214215c827e";
+    sha256 = "ud0tchMJJJklFlGEmeluDcqtjldlsvSecD0F1p0TK+Y=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [
+    llvmPackages.llvm
+    llvmPackages.lld
+  ];
+
+  meta = with lib; {
+    description = "Compiler for the C3 language";
+    homepage = "https://github.com/c3lang/c3c";
+    license = licenses.lgpl3Only;
+    maintainers = with maintainers; [ luc65r ];
+    platforms = platforms.all;
```

Comment:
Can you mark it broken on aarch64 or fix the compile error?

Reply:
i'll report it upstream

Reply:
Thanks @Artturin

Reply:
There's still this https://github.com/c3lang/c3c/issues/257 but the hello world example builds

Reply:
if you want to test this yourself add `boot.binfmt.emulatedSystems = [ "aarch64-linux" ];` to your config and reboot
then
`nix-build '.' -A c3c --system "aarch64-linux"` in nixpkgs

Reply:
add this to run the tests

```
  checkInputs = [ python3 ];

  postPatch = ''
    patchShebangs test/src/tester.py
  '';

  doCheck = true;

  checkPhase = ''
    runHook preCheck
    
    pushd ../resources/testproject
    ../../build/c3c build
    popd

    pushd ../test
    python src/tester.py ../build/c3c test_suite/
    popd

    runHook postCheck
  '';
```

Reply:
There's only this but it doesn't affect whether or not the build completes

```
- test_suite/assert/global_static_assert_not_constant.c3: Failed.
Error (-6): *** stack smashing detected ***: terminated
qemu: uncaught target signal 6 (Aborted) - core dumped
```
so lets remove that for now

Reply:
We dont know if this builds on darwin so lets set platforms to linux only

this can be applied with `git apply file.patch`
```
diff --git a/pkgs/development/compilers/c3c/default.nix b/pkgs/development/compilers/c3c/default.nix
index 2a58fa70808..7d815593a68 100644
--- a/pkgs/development/compilers/c3c/default.nix
+++ b/pkgs/development/compilers/c3c/default.nix
@@ -22,11 +22,30 @@ llvmPackages.stdenv.mkDerivation rec {
     llvmPackages.lld
   ];
 
+  checkInputs = [ python3 ];
+
+  doCheck = true;
+
+  checkPhase = ''
+    runHook preCheck
+    
+    pushd ../resources/testproject
+    ../../build/c3c build
+    popd
+
+    pushd ../test
+    rm test_suite/assert/global_static_assert_not_constant.c3
+    python src/tester.py ../build/c3c test_suite/
+    popd
+
+    runHook postCheck
+  '';
+
   meta = with lib; {
     description = "Compiler for the C3 language";
     homepage = "https://github.com/c3lang/c3c";
     license = licenses.lgpl3Only;
     maintainers = with maintainers; [ luc65r ];
-    platforms = platforms.all;
+    platforms = platforms.linux;
   };
 }
```

Reply:
I am doing all my dev on Intel Darwin if that helps.

Reply:
> I am doing all my dev on Intel Darwin if that helps.

That should mean it will work 

@luc65r don't apply the platforms part of the patch

Reply:
OfBorg will check if we are patient

Reply:
there's no need to remove `test_suite/assert/global_static_assert_not_constant.c3` @lerno fixed it